### ISSUE TITLE
make build fails on full build

### ISSUE THREAD

**Issue by nbx3** [mlan/docker-asterisk#14]
This seems related to issue #6 

I can't work out why the below error is occurring when the links are being read, but setting DOCKER_BUILDKIT=0 fixes the problem - otherwise you can only build base and mini images.

```
git submodule update --init --recursive
mkdir -p sub/autoban/php
ln -f sub/module/phpami/src/Ami.php sub/autoban/php/ami.class.inc
mkdir -p sub/codecs/module
ln -f sub/codecs/download/codec_g723-ast160.so sub/codecs/module/codec_g723.so
mkdir -p sub/codecs/module
#ln -f sub/codecs/download/codec_g729-ast160.so sub/codecs/module/codec_g729.so
DOCKER_BUILDKIT=1 docker build --build-arg DIST=alpine --build-arg REL=3.19 --build-arg PHP_VER=php82 --target full \
--tag mlan/asterisk:latest --tag mlan/asterisk:full .
[+] Building 0.6s (5/16)                                                                                                             docker:default
 => [internal] load build definition from Dockerfile                                                                                           0.0s
 => => transferring dockerfile: 3.31kB                                                                                                         0.0s
 => [internal] load metadata for docker.io/library/alpine:3.19                                                                                 0.4s
 => [internal] load .dockerignore                                                                                                              0.0s
 => => transferring context: 103B                                                                                                              0.0s
 => ERROR [internal] load build context                                                                                                        0.0s
 => => transferring context: 3.70kB                                                                                                            0.0s
 => [mini 1/9] FROM docker.io/library/alpine:3.19@sha256:c5b1261d6d3e43071626931fc004f70149baeba2c8ec672bd4f27761f8e1ad6b                      0.0s
------
 > [internal] load build context:
------
ERROR: failed to solve: invalid link sub/codecs/module/codec_g723.so to unknown path: "sub/codecs/download/codec_g723-ast160.so"
make: *** [Makefile:42: build_full] Error 1
```

**Comment by mlan** [comment]
Hello @nbx3 

I noticed your pull request also related to the codecs. But if the issue mentioned here still persists I want to highlight that the link command e.g.:
```ln -f sub/codecs/download/codec_g723-ast160.so sub/codecs/module/codec_g723.so```
is creating a hard link. Is that an issue on the file system that you are on?

**Comment by nbx3** [comment]
Hi, I'm on Ubuntu 22 and filesystem is ext4, I checked the hard link files and Inode adresses are correct. I tried reming the dest links and the whole sub/* a few times as well and they get recreated correctly by make.
Only way to get around it was to use disable build kit, not sure what strange thing is going on. I'm.using docker v26.0 so perhaps that has some regression, I noticed there's a few point release updates to it already so will update and try again tomorrow

**Comment by mlan** [comment]
@nbx3

The issue was caused by the usage of hard links during image build. The behaviour of the docker builder, which previously was able to handle hard links, changed with version 26 of docker.

This issue was addressed in commit db29412 by files being copied on the host instead of being hard-linked.
